### PR TITLE
Add fallback for host.id in Cloud

### PR DESCRIPTION
**Pull request by christos68k**
This PR changes the semantics behind `host.id` calculation for Cloud.

## Changes

In Cloud, it is possible for `instance_id` to not be retrievable from inside the instance (e.g. security concerns / restricted instance metadata). In this case, the `host.id` can be set from `machine-id` (same steps as non-containerized systems).

Without a fallback, the `must` [wording](https://github.com/open-telemetry/semantic-conventions/blob/40839cbd53939126b213f2b50921b92a107719e0/model/registry/host.yaml#L11) has strong implications for clients and users in cases where the `instance_id` can not be automatically retrieved:

1. Do not automatically populate `host.id`
2. Urge the user to either provide `instance_id` or change the configuration to make it retrievable

A (breaking) alternative to the current approach is to always populate `host.id` from `machine-id`, and use a different field for `instance_id` (e.g. [service.instance.id](https://github.com/open-telemetry/semantic-conventions/pull/312))

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Comment by AlexanderWert** [comment]
@open-telemetry/semconv-system-approvers @open-telemetry/semconv-container-approvers PTAL

File: docs/attributes-registry/host.md [hunk 1]
```diff
@@ -14,7 +14,7 @@
 | `host.cpu.model.name` | string | Model designation of the processor. | `11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz` |
 | `host.cpu.stepping` | int | Stepping or core revisions. | `1` |
 | `host.cpu.vendor.id` | string | Processor manufacturer identifier. A maximum 12-character string. [1] | `GenuineIntel` |
-| `host.id` | string | Unique host ID. For Cloud, this must be the instance_id assigned by the cloud provider. For non-containerized systems, this should be the `machine-id`. See the table below for the sources to use to determine the `machine-id` based on operating system. | `fdbf79e8af94cb7f9e8df36789187052` |
+| `host.id` | string | Unique host ID. For Cloud, this must be the instance_id assigned by the cloud provider. If the instance_id can not be retrieved, this should be the `machine-id`. For non-containerized systems, this should be the `machine-id`. See the table below for the sources to use to determine the `machine-id` based on operating system. | `fdbf79e8af94cb7f9e8df36789187052` |
```
[the remark this answers is not in the document]
Comment (pyohannes) [reply docs/attributes-registry/host.md:17]: I see main use cases for `host.id` as similar to `service.instance.id`, quoted from [here](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource#service-experimental): 
> The ID helps to distinguish instances of the same service that exist at the same time.

`service.instance.id` can be a K8s pod name, but that's not guaranteed. If I want that, I look at [`k8s.pod.name`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/k8s.md).

I'd advocate for a similar understanding of `host.id`.

[the remark this answers is not in the document]
Comment (mx-psi) [reply docs/attributes-registry/host.md:17]: Will file an issue for this, but here are the uses of `host.id` that I was able to find that are not machine-id:

- Azure VM ID ([.NET contrib](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/5efd368788185c4e697542f977ee5ad6818a1bb5/src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetadataResponse.cs#L61), [Python contrib](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/4bf3577fb76480dacca12eecd8ed93cb9fccc274/resource/opentelemetry-resource-detector-azure/README.rst#L59). [Ruby contrib](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/73fa809867dd08c1e3fde7413d88538e0df748bf/resources/azure/lib/opentelemetry/resource/detector/azure.rb#L29), [Collector contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5b66f0e2090390f724a9a37ba0dfc60e66bcae0b/processor/resourcedetectionprocessor/internal/azure/azure.go#L56))
- Azure App Service's `WEBSITE_HOSTNAME` ([Python contrib](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/4bf3577fb76480dacca12eecd8ed93cb9fccc274/resource/opentelemetry-resource-detector-azure/README.rst#L59))
- AWS EC2 instance id ([PHP contrib](https://github.com/open-telemetry/opentelemetry-php-contrib/blob/033ed80036fb0af28419d6c14b69ca6cafc95cde/src/Aws/src/Ec2/Detector.php#L87), [Collector contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5b66f0e2090390f724a9a37ba0dfc60e66bcae0b/processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go#L81))
- GKE Host id, GCE Host id ([Go contrib](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/f7ca3ec39ee91e6c3a2e05f0f12f93f1ce3456da/detectors/gcp/detector.go#L58), [Ruby contrib](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/resources/google_cloud_platform/README.md), [Collector contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5b66f0e2090390f724a9a37ba0dfc60e66bcae0b/processor/resourcedetectionprocessor/internal/gcp/gcp.go#L68))
- Consul node id ([Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5b66f0e2090390f724a9a37ba0dfc60e66bcae0b/processor/resourcedetectionprocessor/internal/consul/consul.go#L74))

[the remark this answers is not in the document]
Comment (mx-psi) [reply docs/attributes-registry/host.md:17]: @christos68k @AlexanderWert PTAL at #600. I want to start with this and see what approvers and maintainers of semconv think

Comment (jsuereth) [inline docs/attributes-registry/host.md:17]: Nit:

```
If the instance_id can not be retrieved, this should be the `machine-id`. For non-containerized systems, this should be the `machine-id`.
```

Should be rewritten for better clarity.

```
If the instance_id can not be retrieved, or for non-containerized systems, this should be the `machine-id`.
```

**Review by mx-psi — COMMENTED** [review]
Filed #739 for the general issue I think needs to be solved before accepting this (see also the unresolved thread above)

**Comment by christos68k** [comment]
Coul;d someone with rights (@mx-psi @AlexanderWert maybe?) to do so reopen this pull request?

**Comment by jsuereth** [comment]
Given activity happening in Entity SIG - I think it's best to "wait" on major changes to resource related attributes.  My own opinion on this Pr is that:

1. Yes, using machine id makes sense as the default for host.id
2. Having cloud-specific ids land in `host.id` seems problematic from a consistency/implementation standpoint.   We need to define mechanisms where we would understand how a cloud-provider Resource detector would interact with a Host-specific resource detector.   This change would force the ability for us to understand merge conflcts and overrides between these two disjoint components.

As such - (while I think machine id should be the default for host.id) I think we need to hold off on this PR until more progress has been made in Entity SiG - particularly the mapping between Resource <-> Entity and conflict resolution.

**Pull request closed without merging**